src/topology/maps.lean

```diff
@@ -13,130 +13,200 @@ local attribute [instance] classical.prop_decidable
 
 variables {α : Type*} {β : Type*} {γ : Type*} {δ : Type*}
 
+section dense_range
+variables [topological_space α] [topological_space β] [topological_space γ]
+          (f : α → β) (g : β → γ)
+
+def dense_range := ∀ x, x ∈ closure (range f)
+
+lemma dense_range_iff_closure_eq : dense_range f ↔ closure (range f) = univ :=
+eq_univ_iff_forall.symm
+
+variables {f}
+
+lemma dense_range_comp (hf : dense_range f) (hg : dense_range g) (cg : continuous g) :
```

Comment:
```suggestion
lemma dense_range.comp (hg : dense_range g) (hf : dense_range f) (cg : continuous g) :
```
the convention for composition is that the order in the assumptions should match the order in `g o f`. Also use a name with dot for easier use.

Reply:
I think the dot name won't do anything, because `dense_range` is not a structure. I could make it a structure, but I was afraid to go overboard with structures for properties of maps. Maybe I'll do it at some point.

Reply:
It's surprising at first but extremely useful: you don't need structures for the dot notation. If you have an assumption `h : dense_range f`, `h.comp` will be understood as `dense_range.comp h` even if `dense_range` is not a structure.